### PR TITLE
Feat: support read from URL for --file command globally 

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

Support read from remote URL and stdin for the following commands:

* `vela up`
* `vela def apply`
* `vela kube apply`
* `vela dry-run -f https://<host>/app.yaml`
* `vela live-diff -f https://<host>/app.yaml`

Remove `vela template context` command as it's no longer used for more than 3 minor versions.

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: references/cli/addon.go [hunk 1]
```diff
@@ -131,12 +131,14 @@ func NewAddonEnableCommand(c common.Args, ioStream cmdutil.IOStreams) *cobra.Com
...
+		Example: `  Enable addon by:
 	vela addon enable <addon-name>
-Enable addon with specify version:
+  Enable addon with specify version:
 	vela addon enable <addon-name> --version <addon-version>
-Enable addon for specific clusters, (local means control plane):
+  Enable addon for specific clusters, (local means control plane):
 	vela addon enable <addon-name> --clusters={local,cluster1,cluster2}
+  Enable addon locally:
+	vela addon enable <your-local-addon-path>
```
Comment (charlie0129) [inline references/cli/addon.go:141]: Also add an example about how to specify parameters?

Comment (wonderflow) [reply references/cli/addon.go:141]: yes, good idea

**Review by Somefive — APPROVED** [review]
LGTM

File: references/cli/addon.go [hunk 2]
```diff
@@ -131,12 +131,16 @@ func NewAddonEnableCommand(c common.Args, ioStream cmdutil.IOStreams) *cobra.Com
...
+  Enable addon with specified args (the args should be defined in addon's parameters):
+	vela addon enable <addon-name> --<my-parameter>=<my-value>
```
Comment (charlie0129) [inline references/cli/addon.go:143]:
```suggestion
	vela addon enable <addon-name> <my-parameter>=<my-value>
```

File: references/cli/addon.go [hunk 3]
```diff
@@ -229,12 +233,16 @@ func NewAddonUpgradeCommand(c common.Args, ioStream cmdutil.IOStreams) *cobra.Co
...
-Upgrade addon with specify version:
+  Upgrade addon with specify version:
 	vela addon upgrade <addon-name> --version <addon-version>
-Upgrade addon for specific clusters, (local means control plane):
+  Upgrade addon for specific clusters, (local means control plane):
 	vela addon upgrade <addon-name> --clusters={local,cluster1,cluster2}
+  Upgrade addon locally:
+	vela addon enable <your-local-addon-path>
+  Upgrade addon with specified args (the args should be defined in addon's parameters):
+	vela addon enable <addon-name> --<my-parameter>=<my-value>
```
Comment (charlie0129) [inline references/cli/addon.go:245]:
```suggestion
	vela addon enable <addon-name> <my-parameter>=<my-value>
```

**Pull request merged**